Lib/mastermind.rb

```diff
@@ -29,20 +29,23 @@ def generate_code
 class Response
   attr_reader :player_input
 
-  def initialize
+  def initialize(new_input)
     @valid_guess = false
-    @player_input = ""
+    @player_input = new_input
   end
 
   def get_input
     puts "Please enter a guess from the available colors: r, g, o, b, y, p"
-    @player_input = gets.chomp
-    return @player_input.downcase.gsub(/[\W]/, "")
+    @player_input = gets.chomp.downcase.gsub(/[\W]/, "")
   end
 
-  def validate_length(guess)
-    if @player_input.length != 4
-      puts "Invalid response, please enter exactly 4 colors"
+  def valid_length?(input = @player_input)
```

Comment:
`input = @player_input` does this do anything when the function is called `@player_input`?

Reply:
Nope, deleted - thanks!